src/components/shape/ammo-shape.js

```diff
@@ -0,0 +1,82 @@
+/* global Ammo,THREE */
+const threeToAmmo = require("three-to-ammo");
+
+var AmmoShape = {
+  schema: {
+    type: { default: "hull", oneOf: ["box", "cylinder", "sphere", "capsule", "cone", "hull", "mesh"] },
+    recenter: { default: false }, //recenter the object3D's geometry
+    autoGenerateShape: { default: true }, //disable if using custom halfExtents or sphereRadius
+    mergeGeometry: { default: true }, //use all geometry on the object to generate the shape. Otherwise, setMesh must be called explicitly
+    halfExtents: { type: "vec3", default: { x: 1, y: 1, z: 1 } },
+    minHalfExtent: { default: 0 },
+    maxHalfExtent: { default: Number.POSITIVE_INFINITY },
+    sphereRadius: { default: NaN },
+    cylinderAxis: { default: "y", oneOf: ["x", "y", "z"] },
+    margin: { default: 0.01 },
+    offset: { type: "vec3", default: { x: 0, y: 0, z: 0 } },
+    orientation: { type: "vec4", default: { x: 0, y: 0, z: 0, w: 1 } }
```

Comment:
User-facing rotations should probably be Eulers, to match A-Frame convention.

Reply:
We never actually changed that for the `shape` component, and so as it stands, this is just matching the old `shape` component schema. Given that I don't think there's been much complaint about it being a quaternion in the `shape`c component, and also if you're adding custom rotations for shapes you're probably not doing it "manually", its probably fine to keep it as  a vec4.

Reply:
I still want to change it, but I don't mind waiting until after this PR is merged. :)